cmd/models/validate.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright © 2023 OpenFGA
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package models
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/spf13/cobra"
+	pb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type validationResult struct {
+	ID        string     `json:"id,omitempty"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+	IsValid   bool       `json:"is_valid"`
+	Error     *string    `json:"error,omitempty"`
+}
+
+func validate(inputModel string) (string, error) {
+	model := &pb.AuthorizationModel{}
+
+	err := protojson.Unmarshal([]byte(inputModel), model)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse input due to %w", err)
+	}
+
+	output := validationResult{
+		IsValid: true,
+	}
+
+	if model.Id != "" {
+		modelID, err := ulid.Parse(model.Id)
+		if err != nil {
+			return "", fmt.Errorf("error parsing model id (ID=%s). Error = %w", model.Id, err)
+		}
+
+		output.ID = model.Id
+		createdAt := time.Unix(int64(modelID.Time()/1_000), 0).UTC() //nolint:gomnd
```

Comment:
little bit concerned that this data may not be exactly identical to the output of the list/get model command.

Reply:
This is more helpful for debugging, so maybe it shouldn't be?

Reply:
Will add created_at to the output of list and get

Reply:
ok in this case, we will keep it as is then.